### ISSUE TITLE
suggested changes/enhancements

### ISSUE THREAD

**Issue by prabbit237** [arendst/Sonoff-Tasmota#2092]
This isn't quite a bug but....There was two things I had issues with on the clock code and think that a couple of extra options would help with.

1) I had installed a 60-LED ring and then realized I had it mounted upside down in a frame (and no, it couldn't be easily rotated at this point. It's mounted to the backing board of a picture frame with the kickstand leg and I glued it down before double-checking the direction *sigh* Also I have it mounted behind and shining through a picture and at times I may want the picture to be landscape, as in the Hubble photo of Saturn I'm currently using, and at other times set in portrait orientation so no matter what, I'd need to be able to rotate the clock.) I added the following code as the first line in Ws2812UpdateHand()

    ```position = (position + 30) % 60;```

This had the effect of rotating the ring by 180 degrees. Changing the 30 would change the rotation and maybe making it a percentage or a number of degrees would be better (this was a special case of where I knew the size and rotation needed so using a constant was easiest for this specific case.)

2) Lack of hour markers. I added the following at the end of the same function:

```
WsColor marker_color = {50, 50, 50};
for (byte i = 0; i < 60; i++) {
   if ((i % 5) == 0) {
      Ws2812UpdatePixelColor(i, marker_color, 1);
  }
}
```

Again, I specified it for my particular case. This could also be turned off or via a command if needed/desired.

**Comment by arendst** [comment]
Ha, so I have to consider every possible physical mismount with an option allowing rotation and mirroring for every second..... ;-)

The hour marker is a better candidate for implementation.

**Comment by prabbit237** [comment]
*laugh*
Well, it wouldn't need to worry about EVERY possible configuration. Basically the line I used could be generalized as:
``void Ws2812UpdateHand(int position, uint8_t index)
{
  position = (position + Settings.light_pixels_offset) % Settings.light_pixels;``

It would just need one line added to the function and a way of setting Settings.light_pixels_offset via commands and it'd default to "0" (I don't see any reason it'd be needed in any other pattern other than the clock so that's why I stuck it into Ws2812UpdateHand(). And put it as the very first line so that the reversal, etc. all still works as.) Then, if I have 60 leds in the ring and wanted to turn the clock, I'd just set Settings.light_pixels_offset to 30 (as I had to do in my case) to flip it or change the offset to 15 (or 45) to change it from landscape to portrait, etc. All of the other math with reversing the direction, etc would work the same as it has been working and it's all up to the end user to figure out if any offset is even needed and, if so, how much. The offset COULD be coded in terms of percentage (so mine would have been 50%) or in terms of angle (i.e. 180 in my case) but it'd be simplest just to make it in terms of "number of pixels." So now instead of going 0, 1, 2, 3...58, 59, my pixels go 30, 31, 32....58, 59, 0, 1....28, 29. If I used 120 pixels and needed to flip it, I'd change the offset to 60, etc, etc, etc :)

BTW, great work on Tasmota. I've been playing around with it a good bit and figuring out all the ins and outs of how to work it and it's been doing pretty well.
